### ISSUE TITLE
Synology DS211j - Sabnzbd "Failed to run package Service"

### ISSUE THREAD

**Issue by psycl0ne1982** [SynoCommunity/spksrc#4133]
_For new Package Requests, see the guidelines_  

### Setup
_Package Name:_ Sabnzbd 
_Package Version:_ 2.3.9-41

_NAS Model:_ Synology DS211j
_NAS Architecture:_ 
_DSM version:_ 6.2.3-25426 Update 2

### Expected behavior
NAS updated Sabnzbd to version 2.3.9-41 and after updating will no longer run, returns error message "Failed to run package Service".

### Actual behavior
If I log into the Synology DSM software and try to manually run Sabnzbd through the Package Center it returns the error "Failed to run package Service". I reset my Synology NAS and then reinstalled all of the packages on it and all was fine but then it downloaded this 2.3.9-41 update again and started failing in the same way with the exact same error. Before this update everything worked smoothly.

The update for 2.3.9-41 suggests it "Fix DSM Start Service" but it appears to have had the exact opposite effect on my Synology NAS.

### Steps to reproduce
_1._ Open Package Centre.
_2._ Install update 2.3.9-41
_3._ Get error message.

#### Package log
_Check Package Center or `/usr/local/{package}/var/`_
```
Insert the package log here
```
#### Other logs
_E.g. `/var/log/messages` or `/var/log/synopkg.log`_
```
Insert log here
```

**Comment by ymartin59** [comment]
Please provide output of command `ls -la /var/packages/sabnzbd/target/var` and content of files `/var/packages/sabnzbd/target/var/sabnzbd_install.log` and `/var/packages/sabnzbd/target/var/sabnzbd.log`

**Comment by Safihre** [comment]
These 2 lines were removed in the `41` update, but I found on my own NAS that they were required to get the package to run.
https://github.com/SynoCommunity/spksrc/pull/4035/files#diff-5d6bd92e6f38c326c2344d53c1dfb20aR26-R27

Although this might be a different problem.

**Comment by psycl0ne1982** [comment]
Hopefully I have done this correctly, really not used to SSH had to google all the Putty commands, sorry if its wrong.


### Output of ls -la /var/packages/sabnzbd/target/var

root@NAS:~# ls -la /var/packages/sabnzbd/target/var
total 44
drwxr-xr-x 5 sc-sabnzbd sabnzbd 4096 Aug 15 21:23 .
drwxr-xr-x 7 sc-sabnzbd sabnzbd 4096 Aug 15 21:31 ..
drwxr-xr-x 2 sc-sabnzbd sabnzbd 4096 Aug  9 07:41 admin
-rw------- 1 sc-sabnzbd sabnzbd 8027 Aug  9 07:20 config.ini
-rw------- 1 sc-sabnzbd sabnzbd 8027 Aug  9 07:20 config.ini.bak
drwxr-xr-x 2 sc-sabnzbd sabnzbd 4096 Aug  8 21:19 logs
-rw-r--r-- 1 root       root    1470 Aug 15 21:31 sabnzbd_install.log
-rw-r--r-- 1 sc-sabnzbd sabnzbd  346 Aug 16 12:23 sabnzbd.log
drwxr-xr-x 2 sc-sabnzbd sabnzbd 4096 Jul 15  2019 scripts



### content of files /var/packages/sabnzbd/target/var/sabnzbd_install.log

Owner: [sc-sabnzbd(user)]
---------------------
         [0] group:administrators:allow:rwxpdDaARWc--:fd--  (level:0)
Granting 'sc-download' group permissions on /volume1/DOWNLOAD/incomplete
ACL version: 1
Archive: has_ACL,is_support_ACL
Owner: [sc-sabnzbd(user)]
---------------------
         [0] group:administrators:allow:rwxpdDaARWc--:fd--  (level:0)
         [1] group:sc-download:allow:rwxpdDaARWcC-:fd--  (level:0)



### content of files /var/packages/sabnzbd/target/var/sabnzbd_install.log

Sun Aug 16 12:23:25 AEST 2020
Starting sabnzbd command env LANG=en_US.UTF-8 /volume1/@appstore/sabnzbd/env/bin/python /volume1/@appstore/sabnzbd/share/SABnzbd/SABnzbd.py -f /volume1/@appstore/sabnzbd/var/config.ini --pidfile /volume1/@appstore/sabnzbd/var/sabnzbd.pid -d
env: /volume1/@appstore/sabnzbd/env/bin/python: No such file or directory

**Comment by Safihre** [comment]
Oke seems something else is wrong than what I thought!

**Comment by ymartin59** [comment]
Installation seems to have failed - looks like pip virtual environment has not been created properly in `/volume1/@appstore/sabnzbd/env/`.

Here is expected content of `/var/packages/sabnzbd/target/var/sabnzbd_install.log` (notice on my system, that file is owned by `sc-sabnzbd` not by `root` - but I am not sure it is an issue)

```
Sun Aug 16 09:32:05 CEST 2020
===> Step preinst. USER=sabnzbd GROUP=sc-download SHARE_PATH=/volume1/downloads
Sun Aug 16 09:32:05 CEST 2020
===> Step postinst. USER=sabnzbd GROUP=sc-download SHARE_PATH=/volume1/downloads
Installing service configuration /var/packages/sabnzbd/conf/sabnzbd.sc
Adding 'sc-sabnzbd' to 'sc-download'
Group Name: [sc-download]
Group Type: [AUTH_LOCAL]
Group ID:   [65537]
Group Members: 
0:[sc-demoservice]
1:[sc-deluge]
2:[sc-sabnzbd]
Configuring /volume1/downloads
Invoke service_postinst
New python executable in /volume1/@appstore/sabnzbd/env/bin/python
Installing setuptools, pip, wheel...
done.
Looking in links: /volume1/@appstore/sabnzbd/share/wheelhouse
Processing /volume1/@appstore/sabnzbd/share/wheelhouse/sabyenc-3.3.5-cp27-none-any.whl
Installing collected packages: sabyenc
Successfully installed sabyenc-3.3.5
```

Is Python 2.7.18 package from SynoCommunity installed?
It is a requirement for package and Package Center should have installed it on its own. If not, install it manually from Package Center or else from https://synocommunity.com/package/python

**Comment by psycl0ne1982** [comment]
I do have Python installed, yes.

It did install it automatically when I first installed Sabnzbd after the reset of the NAS last week.

Python Version: 2.7.18-23 is the version that I have installed.

**Comment by ymartin59** [comment]
Please try again with recent 3.0 update running on Python 3.7. https://synocommunity.com/package/sabnzbd

**Comment by psycl0ne1982** [comment]
Hi ymartin59,

Thanks for your help with this one.

Is there anyway to force the package center in the Synology DSM to update and show the newest versions.

I have uninstalled Python and the previous version of Sabnnzbd that wasn't running however in package center I still only see the old versions. Python 3 version 3.5.8-8 and Sabnzbd version 2.3.9-41.

Not the versions you mention and linked to. I have tried removing Synocommunity as a package source then re-adding it however that still returned the old versions not the current versions, is there any way to force an update?

Thanks ymartin59!
